### PR TITLE
docs(middleware): Make Short-circuiting docs more visual

### PR DESCRIPTION
**Pull request by amarynets**
# Summary of Changes

Update docs of middleware short-circuiting  to make it clearer and more visual

# Related Issues

#1648

### CODE REVIEW

**Comment by amarynets** [comment]
Hi @vytas7 could you look at my PR if all is ok there?

**Comment by vytas7** [comment]
Thanks for this PR @amarynets
I'm a bit swamped with other stuff, I'll take a look as soon as I can!

File: docs/api/middleware.rst [hunk 1]
```diff
@@ -255,7 +256,20 @@ the responder method that the request would have been routed to. However, any
 
 In a similar manner, setting :attr:`~falcon.Response.complete` to ``True`` from
 within a *process_resource* method will short-circuit further request processing
-at that point.
+at that point.
```
Comment (vytas7) [inline docs/api/middleware.rst]: Do not introduce trailing whitespace.

File: docs/api/middleware.rst [hunk 2]
```diff
@@ -255,7 +256,20 @@ the responder method that the request would have been routed to. However, any
 
 In a similar manner, setting :attr:`~falcon.Response.complete` to ``True`` from
 within a *process_resource* method will short-circuit further request processing
-at that point.
+at that point. 
+
+In the example bellow, you can see how request processing will be short-circuit
```
Comment (vytas7) [inline docs/api/middleware.rst]: [bellow](https://dictionary.cambridge.org/dictionary/english/bellow) _(sic)_ :arrow_right: below

Comment (vytas7) [inline docs/api/middleware.rst]: Suggested edit:
request processing will be short-circuited once ``resp.complete`` has been set to ``True``, i.e., the framework will prevent `mob3.process_request`, all `process_resource` methods, as well as the routed responder method from processing the request.

**Comment by amarynets** [comment]
@vytas7 I've updated my PR

File: docs/api/middleware.rst [hunk 3]
```diff
@@ -253,10 +253,25 @@ any remaining *process_request* and *process_resource* methods, as well as
 the responder method that the request would have been routed to. However, any
 *process_response* middleware methods will still be called.
 
-In a similar manner, setting :attr:`~falcon.Response.complete` to ``True`` from
+In a similar manner, setting :attr:`falcon.Response.complete` to ``True`` from
 within a *process_resource* method will short-circuit further request processing
 at that point.
 
+In the example below, you can see how request processing will be short-circuit
```
Comment (kgriffs) [inline docs/api/middleware.rst]: `short-circuit` ➡️ `short-circuited`

File: docs/api/middleware.rst [hunk 4]
```diff
@@ -253,10 +253,25 @@ any remaining *process_request* and *process_resource* methods, as well as
...
+once :attr:`falcon.Response.complete` has been set to
+``True``, i.e., the framework will prevent ``mob3.process_request``, all *process_resource*
+methods, as well as the routed responder method from processing the request,
```
Comment (kgriffs) [inline docs/api/middleware.rst]: `request, however` ➡️ `request. However,`

File: docs/api/middleware.rst [hunk 5]
```diff
@@ -253,10 +253,25 @@ any remaining *process_request* and *process_resource* methods, as well as
...
+however all *process_response* methods will continue to process request::
```
Comment (kgriffs) [inline docs/api/middleware.rst]: This is a bit confusing:

> will continue to process request

I would suggest simply saying:

> will still be called

File: docs/api/middleware.rst [hunk 6]
```diff
@@ -253,10 +253,25 @@ any remaining *process_request* and *process_resource* methods, as well as
...
-In a similar manner, setting :attr:`~falcon.Response.complete` to ``True`` from
+In a similar manner, setting :attr:`falcon.Response.complete` to ``True`` from
 within a *process_resource* method will short-circuit further request processing
 at that point.
 
+In the example below, you can see how request processing will be short-circuited
+once :attr:`falcon.Response.complete` has been set to
+``True``, i.e., the framework will prevent ``mob3.process_request``, all *process_resource*
+methods, as well as the routed responder method from processing the request.
+However all *process_response* methods will still be called::
```
Comment (vytas7) [inline docs/api/middleware.rst]: Let's add a comma here:

> `However, all *process_response* methods will still be <...>`

**Pull request merged**